common/config/persistence.go

```diff
@@ -136,6 +143,50 @@ func (c *Persistence) Validate() error {
 				}
 			}
 		}
+		if ds.ShardedNoSQL != nil {
+			cfg := ds.ShardedNoSQL
+			connections := cfg.Connections
+
+			// validate default shard
+			if cfg.DefaultShard == "" {
+				return fmt.Errorf("ShardedNosql config: defaultShard can not be empty")
+			}
+			if _, found := connections[cfg.DefaultShard]; !found {
+				return fmt.Errorf(
+					"ShardedNosql config: defaultShard (%v) is not defined in connections list", cfg.DefaultShard)
+			}
+
+			// validate history sharding
+			historyShardMapping := cfg.ShardingPolicy.HistoryShardMapping
+			currentShardID := -1
+			for _, shardRange := range historyShardMapping {
+				if _, found := connections[shardRange.Shard]; !found {
+					return fmt.Errorf("ShardedNosql config: Unknown history shard name: %v", shardRange.Shard)
+				}
+				if shardRange.Start != currentShardID+1 {
```

Comment:
I prefer to make start inclusive but end exclusive, because that's a pretty common behavior across different areas.

Reply:
Yeah, I also debated the two options when modeling this part of the config. In a usual scenario, I agree that the start-inclusive & end-exclusive pattern works the best. However, when I was thinking about more arbitrary shard distributions (i.e. creating a mapping for a single shard) it felt like using inclusive start & end was making the config a little more readable.

Thinking about it again though, there are other ways to cleanly handle single shard pinning if it ever becomes a need. So I'll change this to the standard range pattern.